### PR TITLE
filter webhook scope

### PR DESCRIPTION
**Pull request by lhoestq**
it was updating datasets completely on event scopes like "discussion.comment"

### CODE REVIEW

**Review by severo — APPROVED** [review]
Ok, just try to fix the styke

File: services/api/src/api/routes/webhook.py [hunk 1]
```diff
@@ -25,12 +25,16 @@
             "properties": {
                 "headSha": {"type": "string"},
                 "name": {"type": "string"},
-                "type": {"type": "string", "enum": ["dataset", "model", "space"]},
+                "type": {
+                    "type": "string",
+                    "enum": ["repo", "repo.content", "repo.config", "discussion", "discussion.comment"],
```
Comment (severo) [inline services/api/src/api/routes/webhook.py]: it will break if another scope is added, no?

Comment (severo) [reply services/api/src/api/routes/webhook.py]: maybe just let it as "string"

File: services/api/src/api/routes/webhook.py [hunk 2]
```diff
@@ -51,6 +55,7 @@ class MoonWebhookV2Payload(TypedDict):
     event: Literal["add", "remove", "update", "move"]
     movedTo: Optional[str]
     repo: MoonWebhookV2PayloadRepo
+    scope: Literal["repo", "repo.content", "repo.config", "discussion", "discussion.comment"]
```
Comment (severo) [inline services/api/src/api/routes/webhook.py:58]: and here too: only str

File: services/api/src/api/routes/webhook.py (hunk 1 above)
Comment (severo) [reply services/api/src/api/routes/webhook.py]: alternative: only list the accepted values here: if another one, such as "discussion" appears, the JSON validation will fail, and the webhook will be discarded

Comment (severo) [reply services/api/src/api/routes/webhook.py]: also: is "scope" always present (even in "delete")?

Comment (lhoestq) [reply services/api/src/api/routes/webhook.py]: this one was a bad copy paste

File: services/api/src/api/routes/webhook.py (hunk 2 above)
Comment (lhoestq) [reply services/api/src/api/routes/webhook.py:58]: ok let me do the change

**Pull request merged**